version: '3.8'
services:
  backend:
    build: ./backend
    ports:
      - "5000:5000"
    depends_on:
      db:
        condition: service_healthy
    environment:
      DB_HOST: db
      DB_USER: root1
      DB_PASSWORD: ${DB_PASSWORD}
      DB_NAME: my_db
    env_file:
      - ./backend/.env
    networks:
      - my-python-app-network
  frontend:
    build: ./frontend/frontends
    ports:
      - "3000:80"   # Or "80:80" if serving with Nginx
  db:
    image: mysql:8
    environment:
      MYSQL_ROOT_PASSWORD: Krishna8kichu
      MYSQL_DATABASE: my_db
      MYSQL_USER: root1
      MYSQL_PASSWORD: Krishna8kichu
    ports:
      - "3310:3306"
    volumes:
      - db_data:/var/lib/mysql
      - ./backend/init.sql:/docker-entrypoint-initdb.d/init.sql:ro
    networks:
      - my-python-app-network
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      interval: 10s
      timeout: 5s
      retries: 5
  prometheus:
    image: prom/prometheus:latest
    volumes:
      - ./prometheus.yaml:/etc/prometheus/prometheus.yaml:ro    # read-only permissions
    command:
      - '--config.file=/etc/prometheus/prometheus.yaml'
    ports:
      - "9090:9090"
    networks:
      - my-python-app-network
    depends_on:
      - backend
  grafana:
    image: grafana/grafana:latest
    ports:
      - "4000:3000"
    environment:
      - GF_SECURITY_ADMIN_PASSWORD=admin
    volumes:
      - grafana_data:/var/lib/grafana
    depends_on:
      - prometheus
volumes:
  db_data:
  grafana_data:
networks:
  my-python-app-network:
    driver: bridge